dsgrid/registry/project_registry_manager.py

```diff
@@ -206,66 +214,199 @@ def _register_project_and_dimensions(
-                ]
-                to_dim = name_mapping[
-                    (mapping.to_dimension.name, mapping.to_dimension.dimension_type)
-                ]
-
-                mapping_model = run_in_other_dir(
-                    src_dir, MappingTableModel.from_pre_registered_model, mapping, from_dim, to_dim
+            mappings.append(mapping_model)
+
+        mapping_config = DimensionMappingsConfig.load_from_model(
+            DimensionMappingsConfigModel(mappings=mappings),
+            src_dir,
+        )
+        mapping_ids = self._dimension_mapping_mgr.register_from_config(
+            mapping_config, submitter, log_message, context=context, force=force
+        )
+        model.dimension_mappings.base_to_supplemental_references += (
+            self._dimension_mapping_mgr.make_dimension_mapping_references(mapping_ids)
+        )
+        model.dimension_mappings.base_to_supplemental.clear()
+
+    def _register_dimensions_from_models(
+        self,
+        src_dir,
+        dimensions: List,
+        dimension_references,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        dim_model = DimensionsConfigModel(dimensions=dimensions)
+        dims_config = DimensionsConfig.load_from_model(dim_model, src_dir)
+        dimension_ids = self._dimension_mgr.register_from_config(
+            dims_config, submitter, log_message, force=force, context=context
+        )
+        # Order of the next two is required for Pydantic validation.
+        dimension_references += self._dimension_mgr.make_dimension_references(dimension_ids)
+        dimensions.clear()
+
+    def _register_all_in_one_dimensions(
+        self,
+        src_dir,
+        supp_dir,
+        model,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        new_dimensions = []
+        new_mappings = {}
+        dim_type_to_ref = {x.dimension_type: x for x in model.dimensions.base_dimension_references}
+        for dimension_type in model.dimensions.all_in_one_supplemental_dimensions:
+            if dimension_type == DimensionType.TIME:
+                raise DSGInvalidParameter("Cannot have time in all_in_one_supplemental_dimensions")
+            if dimension_type in new_mappings:
+                raise DSGInvalidParameter(
+                    f"{dimension_type} is listed twice in all_in_one_supplemental_dimensions"
                 )
-                mappings.append(mapping_model)
 
-            mapping_config = DimensionMappingsConfig.load_from_model(
-                DimensionMappingsConfigModel(mappings=mappings),
+            dim_ref = dim_type_to_ref[dimension_type]
+            dim_config = self._dimension_mgr.get_by_id(dim_ref.dimension_id)
+            dt_str = dimension_type.value
+            dt_plural = f"all_{dt_str}s"
+            dt_plural_dash = f"all-{dt_str}s"
+            dt_plural_formal = f"All {dt_str[0].upper()}{dt_str[1:]}"
```

Comment:
Isn't it just `f"All {dt_str.title()}s"` here?

Reply:
Right. Forgot about that shortuct.